lib/exec.js

```diff
@@ -20,29 +20,31 @@ function exec({ cmd, args, env, pipe, captureOut, captureErr, ...opts }) {
             env: childEnv,
             ...opts,
         })
-        let output = captureOut || captureErr ? '' : undefined
-
-        child.stdout.on('data', (data) => {
-            if (captureOut) {
-                output += data
-            }
-            if (!pipe) {
-                reporter.dump(data)
-            } else {
-                reporter.pipe(data)
-            }
-        })
 
-        child.stderr.on('data', (data) => {
-            if (captureErr) {
-                output += data
-            }
-            if (!pipe) {
-                reporter.pipeErr(data)
-            } else {
-                reporter.printErr(data)
-            }
-        })
+        let output = captureOut || captureErr ? '' : undefined
```

Comment:
(minor)

Technically this does break the `captureOut` / `captureErr` functionality, because specifying those options should resolve the promise with the output, but with this change if `opts.stdio` is specified the output won't be captured.  

This is probably fine, maybe we should just add a small warning `if (opts.stdio && (captureOut || captureErr) { reporter.warn('A custom stdio was passed to the exec command. 
 Output will not be captured.') }`
 
This might honestly be unnecessary, it could just be documented somewhere, but wanted to point it out.

Reply:
Aha, gotcha -- yeah it's probably a bit unlikely that someone will intentionally use both options, but let me see if I can make this a bit more elegant

Reply:
I think it's fine either way 👍

Reply:
@amcgee so I changed it up a bit -- I looked closer into how `spawn` works, and it seems like `process.stderr` and `subprocess.stdout` are both [`null` if the `stdio` option is anything other than the default](https://nodejs.org/api/child_process.html#subprocessstderr) = `'pipe'` (but different from the `pipe` option we use for `exec`)

Given that, I see the `captureOut` functionality as already broken before these changes if the `stdio` option is used, and these changes actually catch that error 🙂